drivers/SmartThings/matter-pump/src/embedded-cluster-utils.lua

```diff
@@ -0,0 +1,52 @@
+local MatterDriver = require "st.matter.driver"
+local clusters = require "st.matter.clusters"
+
+local log = require "log"
+local utils = require "st.utils"
+
+-- Include driver-side definitions when lua libs api version is < 10
+local version = require "version"
+if version.api < 10 then
+  clusters.PumpConfigurationAndControl = require "PumpConfigurationAndControl"
+end
+
+local embedded_cluster_utils = {}
+
+local embedded_clusters = {
+  [clusters.PumpConfigurationAndControl.ID] = clusters.PumpConfigurationAndControl
+}
+
+function embedded_cluster_utils.get_endpoints(device, cluster_id, opts)
+  -- If using older lua libs and need to check for an embedded cluster feature,
+  -- we must use the embedded cluster definitions here
+  if version.api < 10 and embedded_clusters[cluster_id] ~= nil then
+    local embedded_cluster = embedded_clusters[cluster_id]
+    local opts = opts or {}
+    if utils.table_size(opts) > 1 then
+      log.warn_with({hub_logs = true}, "Invalid options for get_endpoints")
+      return
+    end
+    local clus_has_features = function(clus, feature_bitmap)
+      if not feature_bitmap or not clus then return false end
+      return embedded_cluster.are_features_supported(feature_bitmap, clus.feature_map)
+    end
+    local eps = {}
+    for _, ep in ipairs(device.endpoints) do
+      for _, clus in ipairs(ep.clusters) do
+        if ((clus.cluster_id == cluster_id)
+              and (opts.feature_bitmap == nil or clus_has_features(clus, opts.feature_bitmap))
+              and ((opts.cluster_type == nil and clus.cluster_type == "SERVER" or clus.cluster_type == "BOTH")
+                or (opts.cluster_type == clus.cluster_type))
```

Comment:
Could be simplified to:

`and (clus.cluster_type == opts.cluster_type or clus.cluster_type == "SERVER" or clus.cluster_type == "BOTH")`

Reply:
I think this is set up this way in case the goal is to have `opts.cluster_type == "CLIENT"` as an option, but to default to just returning server cluster endpoint if no option is provided. I don't expect we would use the client option, but I will keep it like this to maintain parity with the scripting-engine definition of this function.